profiler/src/ProfilerEngine/Datadog.Linux.ApiWrapper/functions_to_wrap.c

```diff
@@ -403,74 +409,114 @@ int dladdr(const void* addr_arg, Dl_info* info)
 /* Function pointers to hold the value of the glibc functions */
 static int (*__real_execve)(const char* pathname, char* const argv[], char* const envp[]) = NULL;
 
-int execve(const char* pathname, char* const argv[], char* const envp[])
+__attribute__((visibility("hidden")))
+int ShouldCallCustomCreatedump(const char* pathname, char* const argv[])
 {
-    check_init();
+    if (crashHandler == NULL || pathname == NULL)
+    {
+        return 0;
+    }
+
+    size_t length = strlen(pathname);
 
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wtautological-compare"
-    if (crashHandler != NULL && pathname != NULL)
+    if (length < 11 || strcmp(pathname + length - 11, "/createdump") != 0)
     {
-        size_t length = strlen(pathname);
+        return 0;
+    }
 
-        if (length >= 11 && strcmp(pathname + length - 11, "/createdump") == 0)
+    // datadog_crashtracking is set to identify actual crash to dump generation requests (ex: dotnet-dump)
+    for (int i = 0; argv[i] != NULL; i++)
+    {
+        if (strncmp(argv[i], datadogCrashMarker, strlen(datadogCrashMarker)) == 0)
```

Comment:
OPT: might be worth checking that it follows `--name`, though I don't see how it could be present otherwise